src/components/extendedFilter/filterBy.tsx

```diff
@@ -0,0 +1,82 @@
+import ReorderIcon from '@mui/icons-material/Reorder';
+import { IconButton } from '@mui/material';
+import './extendedFilter.scss';
+import { useState } from 'react';
+import { CourseData } from '../../types/components/componentType';
+
+const FilterBy: React.FC = () => {
+  const [isFilterVisible, setFilterVisible] = useState<boolean>(false);
+  const [filteredCourses, setFilteredCourses] = useState<CourseData[]>([]);
+
+  const toggleFilter = () => {
+    setFilterVisible(!isFilterVisible);
+  };
```

Comment:
```suggestion
  const toggleFilter = () => {
    setFilterVisible((prevValue) => !prevValue);
  };
```
I know that in previous ticket, I have proposed you to use this simplier construction. But I have recently read that correct way of changing state is - **immediately** changing state with using callback function, when we click on button.

Reply:
Okay, thanks. I`ll change it